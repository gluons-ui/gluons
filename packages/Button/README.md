# Mandala UI Button
Mandala is a React component library that utilizes a declarative CSS.

### Installation

Check the [Monorepo README](https://github.com/mandala-ui/mandala) for installation.

### Usage
if using the whole library:
```js
import { Button } from 'mandala';
```
if only using this component:
```js
import Button from '@mandala-ui/button';
```

### Props
* `children` - Whatever node you would like in your button. Use this to add icons or customize text color inside.
* `color` - color string for the background color of the button
* `disabled` - disables clicking and reduces the opacity by 60%
* `onClick` - click callback handler
* `pill` -  gives the button 100% radius
* `radius` - border radius, from 0-4, will not work if `pill` is `true`

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| children | node     | null         | -          |
| color    | string   | black        | -          |
| disabled | boolean  | false        | -          |
| onClick  | func     | null         | -          |
| pill     | boolean  | false        | -          |
| radius   | number   | 0            | -          |

![mandala-bottom](https://user-images.githubusercontent.com/1824267/38281372-32114e5c-375f-11e8-9fbe-e5191b96429c.png)
