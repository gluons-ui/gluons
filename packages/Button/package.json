{
  "author": "Lucas K Allmon",
  "description": "Mandala UI Kit Button Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user", "ux"
  ],
  "license": "MIT",
  "main": "lib/Button.js",
  "module": "es/Button.js",
  "name": "@mandala-ui/button",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "1.0.2",
  "dependencies": {
    "lodash.clamp": "^4.0.3"
  }
}
