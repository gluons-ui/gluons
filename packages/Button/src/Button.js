import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class Button extends PureComponent {
  render() {
    const {
      children,
      color,
      disabled,
      onClick,
      pill,
      radius,
    } = this.props;
    const options = `bg-${color} ${pill ? 'br-pill' : `br${radius}`} ${disabled ? 'opacity-40' : 'pointer'}`;
    return (
      <button
        disabled={disabled}
        className={`${options} bn outline-o ph3 pv2 white`}
        onClick={onClick}
        style={{ fontFamily: 'inherit' }}
      >
        {children || null}
      </button>
    );
  }
}

Button.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  pill: PropTypes.bool,
  radius: PropTypes.number,
};

Button.defaultProps = {
  children: null,
  color: 'gray',
  disabled: false,
  onClick: () => {},
  pill: false,
  radius: 0,
};

export default Button;
