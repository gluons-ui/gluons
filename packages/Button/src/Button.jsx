import React from 'react';
import PropTypes from 'prop-types';
import clamp from 'lodash/clamp';

const Button = ({
  children,
  color,
  disabled,
  onClick,
  pill,
  radius,
}) => {
  const isDisabled = `${disabled ? 'o-40' : 'pointer'}`;
  const isPill = `${pill ? 'br-pill' : `br${clamp(radius, 0, 4)}`}`;
  const options = `bg-${color} ${isPill} ${isDisabled}}`;
  return (
    <button
      disabled={disabled}
      className={`button bn outline-0 ph3 pv2 white ${options}`}
      onClick={onClick}
      style={{ fontFamily: 'inherit' }}
    >
      {children || null}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  pill: PropTypes.bool,
  radius: PropTypes.number,
};

Button.defaultProps = {
  children: null,
  color: 'gray',
  disabled: false,
  pill: false,
  radius: 0,
};

export default Button;
