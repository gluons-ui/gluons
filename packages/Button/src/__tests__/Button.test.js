import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('<Button />', () => {
  it('renders defaults correctly', () => {
    const comp = renderer.create(
      <Button>
        Button
      </Button>,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
