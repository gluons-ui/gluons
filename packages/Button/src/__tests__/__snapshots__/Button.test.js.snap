// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders defaults correctly 1`] = `
<button
  className="bg-gray br0 pointer bn outline-o ph3 pv2 white"
  disabled={false}
  onClick={undefined}
  style={
    Object {
      "fontFamily": "inherit",
    }
  }
>
  Button
</button>
`;
